src/limitedmap.h

```diff
@@ -66,8 +66,7 @@ class limitedmap
     }
     void update(const_iterator itIn, const mapped_type& v)
     {
-        // TODO: When we switch to C++11, use map.erase(itIn, itIn) to get the non-const iterator.
-        iterator itTarget = map.find(itIn->first);
+        iterator itTarget = map.erase(itIn, itIn);
```

Comment:
The access is constant time versus the lookup,  so it is an improvement (assuming this is actually a performance bottleneck anyhow).
I think a SO link and explanation would be more than fine to appropriate its existence.